osquery/tables/system/darwin/usb_devices.cpp

```diff
@@ -26,6 +26,14 @@ std::string decodeUSBBCD(uint16_t bcd) {
   return std::to_string(major) + "." + std::to_string(minor);
 }
 
+// Decoding the class/usbclass returns.
+std::string decodeUSBBCDToInt(uint16_t bcd) {
```

Comment:
Could you please detail what you're trying to achieve here?

Reply:
Yeah, I spent ages trying to get the correct numbers out of what getNumIOKitProperty() returned, and how it's encoded going by https://beyondlogic.org/usbnutshell/usb5.shtml#DeviceDescriptors, and eventually made that function to get them in the right 0-255 range like the linux table returned for the same devices.

Reply:
```
  r["class"] = std::to_string(getNumIOKitProperty(details, "bDeviceClass") & 0xFF);
  r["subclass"] = std::to_string(getNumIOKitProperty(details, "bDeviceSubClass") & 0xFF);
```

Would this achieve the same thing for you?